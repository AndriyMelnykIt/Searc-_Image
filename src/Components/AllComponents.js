export { default as HomePage } from './HomePage';
export { default as PageNotFound } from './PageNotFound';
export { default as Picture } from './Picture';
export { default as Header } from './Header';
export { default as Photos } from './Photos';
export { default as Spinner } from './Spinner';
export { default as Pages } from './Pages';
export { default as SearchPage } from './SearchPage';
export { default as SearchHistory } from './SearchHistory';
export { default as PhotosPage } from './PhotosPage';

