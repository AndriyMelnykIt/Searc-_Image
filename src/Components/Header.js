import React from 'react';
import {Navbar, Container, FormControl} from "react-bootstrap";

const Header = () => {
    return (
        <div>
            <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
                <Container fluid>
                    <FormControl
                        placeholder="Поле пошуку"
                        aria-label="Username"
                    />
                </Container>
            </Navbar>
        </div>
    );
};

export default Header;