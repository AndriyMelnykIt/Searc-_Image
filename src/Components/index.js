export { default as Homepage } from './Homepage';
export { default as PageNotFound } from './PageNotFound';
export { default as Picture } from './Picture';
