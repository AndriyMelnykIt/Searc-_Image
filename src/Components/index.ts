export { default as HomePage } from './HomePage';
export { default as Picture } from './Walty photos/Picture';
export { default as Header } from './Header';
export { default as Photos } from './Walty photos/Photos';
export { default as Spinner } from './Spinner';
export { default as Pages } from './Pages';
export { default as SearchPage } from './Walty photos/SearchPage';
export { default as SearchHistory } from './Walty photos/SearchHistory';
