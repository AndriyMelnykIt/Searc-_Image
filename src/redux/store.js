import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducer from './ducks/reducer';

import { photos } from './ducks/ducksGeneration';

const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();
const middleware = [sagaMiddleware];

const store = createStore(
	reducer,
	reduxDevtools(applyMiddleware(...middleware))

);

sagaMiddleware.run(photos);

export default store;