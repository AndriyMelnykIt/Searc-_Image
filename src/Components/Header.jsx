import React from 'react';
import {Nav, Navbar} from 'react-bootstrap';
import {Link} from 'react-router-dom';
import {StyledHeader} from '../styled/Header';

const Header = () => {
    return (
        <>
            <StyledHeader>
                <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
                    <Navbar.Brand>
                        Walty Photos
                    </Navbar.Brand>
                    <Nav>
                        <Nav.Link><Link to='/'>Home</Link></Nav.Link>
                        <Nav.Link><Link to='/photo'>Photo</Link></Nav.Link>
                    </Nav>
                </Navbar>
            </StyledHeader>
        </>
    );
};

export default Header;

