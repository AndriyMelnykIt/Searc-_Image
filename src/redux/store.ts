import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import { photos, rootReducer } from './ducks/WaltyPhotos';

const sagaMiddleware = createSagaMiddleware();
const middleware = [sagaMiddleware];

const WallyReducer = combineReducers({
    rootReducer
});

export const store = createStore(
    WallyReducer,
    composeWithDevTools(applyMiddleware(...middleware))
);

sagaMiddleware.run(photos);