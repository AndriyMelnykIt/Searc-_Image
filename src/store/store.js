import { createStore } from 'redux';
import pictureReducer from './ducks/duck.generator';



export const store = createStore(pictureReducer);