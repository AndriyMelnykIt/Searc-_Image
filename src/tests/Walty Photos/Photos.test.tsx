import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';

import { Photos } from '../../Components/index';

import { store } from '../../redux/store';

configure({adapter: new Adapter()});

const photos: any = {
    photo: '',
}

let wrapper: any;

describe('Photos', () => {

    beforeAll(() => {
        wrapper = shallow(<Provider store={store}><Photos photos={photos} loading={photos}/></Provider>);
    });

    it('should take a snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
