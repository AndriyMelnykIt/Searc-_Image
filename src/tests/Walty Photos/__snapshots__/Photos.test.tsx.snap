// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos should take a snapshot 1`] = `ShallowWrapper {}`;
